### DISCUSSION TITLE
Firefox right-click "Open Link in New Private Window" seems to have broken as of TabMixPlus v1.22.0

### DISCUSSION THREAD

**Discussion by ckriger** [onemen/TabMixPlus#316] [General]
As the title says, "Open Link in New Private Window" used to work fine before the latest TabMixPlus update, and still works fine if I disable TabMixPlus as a test, but otherwise nothing happens now if I right-click a link and try to use that option.

Any ideas?

**Comment by onemen** [comment]
@ckriger, 
Thank you for reporting this issue.

What is you Firefox version and operating system?
Do you see any message in the `Browser Console` when trying to use `Open Link in New Private Window`
Does it happened with all links?

Attach a file with you Tab Mix Plus preferences.

**Reply by ckriger** [reply]
I am running Firefox 129.0 (64-bit) on Windows 10, and I have attached my prefs file.

[TMPpref.txt](https://github.com/user-attachments/files/16555883/TMPpref.txt)

I do get the following error in the Browser Console when trying to open seemingly any/all links with that "Open Link in New Private Window" option:

```
Uncaught ReferenceError: win is not defined
    openLinkInPrivateWindow chrome://tabmixplus/content/changecode.js line 222 > eval:2
    <anonymous> chrome://browser/content/browser-context.js:50
2 changecode.js line 222 > eval:2:22
```


Hopefully that helps you track it down but let me know if you need anything else, thanks!

Curious if it "just works" for you?

**Reply by onemen** [reply]
I can not reproduce it on Windows 11.

I have to investigate more...

**Reply by onemen** [reply]
go to your Browser console
enter the command `nsContextMenu.prototype.openLinkInPrivateWindow.toString()`
click enter.
post the result here

If you can not see command line in your `Browser Console`, close it.
set in about:config `devtools.chrome.enabled` to `true` and open `Browser Console` again

**Reply by ckriger** [reply]
Here's that result:
```
'openLinkInPrivateWindow() {
    this.window.openLinkIn(
      this.linkURL,
      "window",
      this._openLinkInParameters({ private: true })
    );
  }'

```

And thanks for the clear instructions on how to turn on the command line btw, I did need to enable that.  :)

**Comment by onemen** [comment]
It is possible that Firefox uses old Tab Mix version from cache

Open Firefox
Remove Tab Mix Plus
Close Firefox
Open Firefox again
Install [Tab Mix Plus version 1.2.2](https://github.com/onemen/TabMixPlus/releases/download/v1.22.0/tab_mix_plus-1.22.0.xpi)

**Reply by ckriger** [reply]
Okay, I did the uninstall/reinstall sequence, and that does seem to have fixed the problem.  Just one of those weird things I guess, but thanks!
